planet/specs.py

```diff
@@ -36,10 +34,53 @@
 HARMONIZE_TOOL_TARGET_SENSORS = ['Sentinel-2']
 BAND_MATH_PIXEL_TYPE = ('Auto', '8U', '16U', '16S', '32R')
 BAND_MATH_PIXEL_TYPE_DEFAULT = 'Auto'
+PRODUCT_BUNDLES = None
 
 LOGGER = logging.getLogger(__name__)
 
 
+class FetchBundlesSpecError(Exception):
+    """Custom exception for errors fetching the product bundles spec."""
+    pass
+
+
+class _LazyBundlesLoader:
+    """Lazy load the product bundles spec from the API."""
+
+    def __getitem__(self, key):
+        bundles_spec_url = "https://api.planet.com/compute/ops/bundles/spec"
+        retries = 2
+        cache = getattr(self, "cache", None)
+        if cache is None:
+            for attempt in range(1, retries + 1):
+                try:
+                    response = httpx.get(bundles_spec_url)
+                    response.raise_for_status()
+                    break
+                except:  # noqa: E722
+                    if attempt == retries:
+                        raise FetchBundlesSpecError(
+                            "Unable to fetch product bundles spec from API to perform client side validation on item types and product bundles. Please retry!"
```

Comment:
```suggestion
                            "Unable to fetch product bundles spec. Please retry!"
```
Will users always want to fetch the bundles spec for validation?  This MR looks like it also supports showing the bundles spec for discovery.

Reply:
Ah, good catch. I didn't update this message after adding the new cli commands. However, users don't have a choice lol the spec has to be fetched for any client side validation OR item type/bundle discovery. Maybe it's overkill to mention item types in the error message though.

Reply:
updated the language, lmk what you think